src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java

```diff
@@ -0,0 +1,194 @@
+            } else {
+                sessionJSON.put("name", session.getName());
+            }
+
+            if (session.getDevice() == null || session.getDevice().isEmpty()) {
+                sessionJSON.put("browser", session.getBrowser());
+            } else {
+                sessionJSON.put("browser", session.getDevice());
+            }
+            sessionJSON.put("os", session.getOs());
+            sessionJSON.put("osVersion", session.getOsVersion());
+            sessionJSON.put("status", session.getBrowserStackStatus());
+
+            if (session.getBrowserStackStatus().equals(session.getStatus())) {
+                sessionJSON.put("userMarked", "UNMARKED");
+            } else {
+                sessionJSON.put("userMarked", session.getStatus());
+            }
+
+
+            // Condition which shouldn't occur if the build is not being reused elsewhere.
+            // But if it happens, the following condition will handle the scenario where
+            // duration is null or empty (running session)
+            if (session.getStatus().equals("running")) {
+                sessionJSON.put("duration", "-");
+            } else {
+                sessionJSON.put("duration", session.getDuration());
+            }
+
+            sessionJSON.put("createdAt", session.getCreatedAt());
+            sessionJSON.put("url", session.getPublicUrl() + "&source=jenkins");
+            sessionsCollection.add(sessionJSON);
+        }
+
+        return sessionsCollection;
+    }
+
+    private static class SessionsSortingComparator implements Comparator<JSONObject> {
+
+        @Override
+        public int compare(JSONObject sessionOne, JSONObject sessionTwo) {
+            // possible values for user_marked: failed, passed and UNMARKED, thus changing all to lowercase
+            String sessionOneUserMarked = sessionOne.getString("userMarked").toLowerCase();
+            String sessionTwoUserMarked = sessionTwo.getString("userMarked").toLowerCase();
+            int userMarkedStatusComparator = sessionOneUserMarked.compareTo(sessionTwoUserMarked);
+
+            // ascending with `user marked status` but descending with `created at`
+            if (userMarkedStatusComparator == 0) {
+                Date sessionOneDate = (Date) sessionOne.get("createdAt");
+                Date sessionTwoDate = (Date) sessionTwo.get("createdAt");
+                int createdAtComparator = sessionOneDate.compareTo(sessionTwoDate);
+
+                return createdAtComparator == 0
+                        ? userMarkedStatusComparator
+                        : (createdAtComparator > 0 ? -1 : 1);
+            }
+            return userMarkedStatusComparator;
+        }
+    }
+
+    public List<JSONObject> getResult() {
+        return result;
+    }
+
+    public List<JSONObject> getResultMeta() {
+        return resultMeta;
+    }
+
+    public String getBuildName() {
+        return buildName;
+    }
+
+    public void setBuildName(String buildName) {
```

Comment:
Are there any chances buildName can be null? Like, what will be the value if there is `Untitled Build` will the value be `Untitled Build`?

Reply:
nope. thats a mandatory thing and it will be fetched from jenkins' `BUILD_TAG` env var...